server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java

```diff
@@ -32,121 +32,32 @@
-import org.opensearch.action.delete.DeleteResponse;
-import org.opensearch.action.index.IndexRequest;
-import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.bulk.BulkItemResponse;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.bulk.TransportBulkAction;
 import org.opensearch.action.support.ActionFilters;
-import org.opensearch.action.support.AutoCreateIndex;
-import org.opensearch.action.support.TransportActions;
-import org.opensearch.action.support.single.instance.TransportInstanceSingleOperationAction;
-import org.opensearch.cluster.ClusterState;
-import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.WriteRequest;
 import org.opensearch.cluster.metadata.Metadata;
-import org.opensearch.cluster.routing.PlainShardIterator;
-import org.opensearch.cluster.routing.ShardIterator;
-import org.opensearch.cluster.routing.ShardRouting;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.collect.Tuple;
 import org.opensearch.common.inject.Inject;
-import org.opensearch.common.logging.DeprecationLogger;
-import org.opensearch.common.settings.Settings;
-import org.opensearch.common.xcontent.XContentHelper;
 import org.opensearch.core.action.ActionListener;
-import org.opensearch.core.common.bytes.BytesReference;
-import org.opensearch.core.common.io.stream.NotSerializableExceptionWrapper;
-import org.opensearch.core.common.io.stream.StreamInput;
-import org.opensearch.core.index.shard.ShardId;
-import org.opensearch.core.rest.RestStatus;
-import org.opensearch.core.xcontent.MediaType;
-import org.opensearch.index.IndexNotFoundException;
-import org.opensearch.index.IndexService;
-import org.opensearch.index.IndexSettings;
-import org.opensearch.index.engine.VersionConflictEngineException;
-import org.opensearch.index.shard.IndexShard;
-import org.opensearch.index.shard.IndexingStats.Stats.DocStatusStats;
-import org.opensearch.indices.IndicesService;
 import org.opensearch.tasks.Task;
-import org.opensearch.threadpool.ThreadPool;
-import org.opensearch.threadpool.ThreadPool.Names;
 import org.opensearch.transport.TransportService;
-import org.opensearch.transport.client.node.NodeClient;
-
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
-
-import static org.opensearch.ExceptionsHelper.unwrapCause;
-import static org.opensearch.action.bulk.TransportSingleItemBulkWriteAction.toSingleItemBulkRequest;
-import static org.opensearch.action.bulk.TransportSingleItemBulkWriteAction.wrapBulkResponse;
 
 /**
- * Transport action for updating an index
+ * Performs the update operation by delegating to {@link TransportBulkAction} with a single update operation.
  *
  * @opensearch.internal
  */
-public class TransportUpdateAction extends TransportInstanceSingleOperationAction<UpdateRequest, UpdateResponse> {
```

Comment:
Since we're changing this class to inherit from `HandledTransportAction` instead of `TransportInstanceSingleOperationAction` I think we can also remove the latter class entirely. Looks like this is the only place where it's used: https://github.com/search?q=org%3Aopensearch-project%20TransportInstanceSingleOperationAction&type=code

Reply:
Nice, thank you!